PanoptesNetClient/PanoptesNetClient/IRequest.cs

```diff
@@ -7,9 +7,10 @@ namespace PanoptesNetClient
     public interface IRequest
     {
         string Endpoint { get; }
+        string Resource { get; }
 
         IRequest ById(string id);
         IRequest WithArgs(NameValueCollection query);
-        void BuildEndpoint();
+        string BuildEndpoint();
```

Comment:
I don't think this should be a part of the interface. Because you already have the property `Endpoint` I don't see why you would need to force the implementing classes to have a helper method, that should just be a part of the abstraction